Update Integration Documentation (#1365)
hashicorp/terraform-website: content/source/docs/cloud/integrations/service-now/index.html.md

# Terraform ServiceNow Service Catalog Integration Setup Instructions

-> **Integration version:**  v2.0.0

> -> **Note:** The ServiceNow Catalog integration is designed for use by Terraform Enterprise customers. We do not currently recommend using it with the SaaS version of Terraform Cloud.

Reviewer: Let's remove this, since we'll be supporting this going forward

Author: Changed this as a note to v1 docs

# Terraform ServiceNow Service Catalog Integration Setup Instructions

-> **Integration version:**  v2.0.0

> -> **Note:** The v1.x integrations have been deprecated, but documentation for existing installations can be found [here](service-now-v1/index.html)

Reviewer 2: ```suggestion
-> **Note:** The v1.x integrations have been deprecated, but you can access documentation for existing installations [here](service-now-v1/index.html)
```

-> **Integration version:**  v2.0.0

> -> **Note:** v1 has been deprecated but can be found [here](service-now-v1/index.html)

Reviewer: `The v1.x integrations have been deprecated, but documentation for existing installations can be found [here]` maybe? Just a minor tweak

Author: Updated

- An account on a [Terraform Enterprise](https://www.hashicorp.com/products/terraform/) instance.
- A ServiceNow instance or developer instance. You can request a ServiceNow developer instance at [developer.servicenow.com](https://developer.servicenow.com/).
- A [supported version control system](../../vcs/index.html#supported-vcs-providers) (VCS) with read access to repositories with Terraform configuration.

It has been tested on the following ServiceNow server versions:

- Madrid
- New York
- Orlando

It requires the following ServiceNow Plugins as dependencies for the Terraform ServiceNow App:

- Flow Designer support for the Service Catalog
- ServiceNow IntegrationHub Action Step
- ServiceNow IntegrationHub Starter Pack

## Obtaining the ServiceNow Integration

Before beginning setup, you must obtain a copy of the Terraform ServiceNow Catalog integration software.

> This can be added to your ServiceNow instance from the [ServiceNow Store](https://store.servicenow.com/).

Reviewer 2: ```suggestion
You can add this to your ServiceNow instance from the [ServiceNow Store](https://store.servicenow.com/).
```

## Installing the ServiceNow Integration

#### Enable Polling Workers (Recommended)

The integration includes 3 ServiceNow Scheduled Flows to poll the Terraform Enterprise API using ServiceNow Outbound HTTP REST requests. By default, all flows schedules are set to 5 minutes. These can be customized inside the ServiceNow Server Studio:

1. Select the Worker Poll Run State Flow.

> 2. Adjust Repeat Intervals

Reviewer 2: These should all be `1. `

3. Click "Done"
4. Click "Save"
5. Click "Activate"

##### Worker Poll Apply Run

This worker approves runs for any workspaces that have finished a Terraform plan and are ready to apply their changes. It also adds a comment on the request item for those workspaces notifying that a run has been triggered.

##### Worker Poll Destroy Workspace

> This worker looks for any records in the Terraform ServiceNow table that is marked for deletion with the value `is_destroyable` set to true. It then checks the status of the workspace to ensure it is ready to be deleted. Once the destroy run has been completed, this work will send the delete request for the workspace to Terraform.

Reviewer 2: ```suggestion
This worker looks for any records in the Terraform ServiceNow table that are marked for deletion with the value `is_destroyable` set to true. It then checks the status of the workspace to ensure it is ready to be deleted. Once the destroy run has been completed, this work will send the delete request for the workspace to Terraform.
```

-> **Roles Required:** `admin` or `x_terraform.config_user`

1. From the ServiceNow Service Management Screen.

> 1. Using the left-hand navigation, open the configuration table for the integration to manage the Terraform Enterprise connection.

Reviewer 2: Should these be indented? I assume they're actions you take from the service now management screen?

Reviewer 2: Also, I'm not sure if we're officially lifting the restriction with using SNow with Terraform Cloud, but from convos with PMs it seemed like they were open to it. Probably out of scope for this PR but we might want to consider changing references to TFE to Cloud instead.

`tf_vcs_record` | Manage ServiceNow Terraform VCS Repositories Table Records
`tf_workspace` | Resources for Terraform Workspace API Requests

### Example Service Catalog Flows and Actions

> The ServiceNow Example Flows can be found in the ServiceNow Studio > Flows. By default, the Flows execute upon submitting an order request for the various catalog items. Admins can modify the Flows and Actions to wait on an approval action, include approval rules, and specify approver groups.

Reviewer 2: Are Flows and Flows and Actions really proper nouns? Other places in this PR they are lower case.

Author: Fixed! (Hopefully 😅 )
